logstash-core/src/main/java/org/logstash/ackedqueue/ext/JRubyWrappedAckedQueueExt.java

```diff
@@ -78,18 +80,18 @@ public JRubyAckedQueueExt rubyGetQueue() {
 
     public void close() throws IOException {
         queue.close();
-        isClosed.set(true);
     }
 
     @JRubyMethod(name = {"push", "<<"})
     public void rubyPush(ThreadContext context, IRubyObject event) {
-        checkIfClosed("write");
         queue.rubyWrite(context, ((JrubyEventExtLibrary.RubyEvent) event).getEvent());
     }
 
     @JRubyMethod(name = "read_batch")
     public IRubyObject rubyReadBatch(ThreadContext context, IRubyObject size, IRubyObject wait) {
-        checkIfClosed("read a batch");
+        if (queue.isClosed()) {
```

Comment:
`Queue#readBatch` is a magic method, it is `synchronized`, also reentrantly locked which I have a _concern_ about moving closed check into `Queue#readBatch`.
The `JRubyWrappedAckedQueueExt` was emulating `isClosed` by synchronizing its atomic `closed` with `Queue#closed`. With current change, we decrease a risk of synchronization delay, especially since our `input` is `volatile`. As a small improvement with recent commit, moved closed check right before `Queue@readBatch` call in `JRubyAckedQueueExt#readBatch`.